### DISCUSSION TITLE
Connection terminated after a few seconds of streaming over the internet (but not on LAN)

### DISCUSSION THREAD

**Discussion by sigmondkukla** [LizardByte/community#182] [Sunshine]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your issue described in the documentation?

- [X] I have read the documentation

### Is your issue present in the nightly release?

- [X] This issue is present in the nightly release

### Describe the Bug

I have set up Sunshine on my gaming PC and Moonlight on both a Chromebook and Android phone. I am able to connect and stream on LAN without issues, but when I try to access it over the internet, it connects and I am able to move the mouse as as expected but only for about 10 seconds.
After that, the stream ends and a variation of a Connection Terminated error appears, depending on what platform Moonlight is installed on.

### Expected Behavior

On WAN / remote / internet, I should be able to maintain a stream just as I can on LAN.

### Additional Context

This seems somewhat similar to this issue/discussion: https://github.com/LizardByte/Sunshine/discussions/927
However, my logs do not match.

I have the following ports forwarded (both TCP and UDP) in my router. Notably not using uPNP, is that an issue?
47989 47984 48010 47998 47999 47800

### Host Operating System

Windows

### Operating System Version

Windows 11 22H2 (25300.1000)

### Architecture

64 bit

### Sunshine commit or version

0.18.4 and nightly

### Package

Windows - installer

### GPU Type

Nvidia

### GPU Model

GeForce RTX 3060 Ti

### GPU Driver/Mesa Version

528.49

### Config

```Shell
sunshine.conf appears to be empty (0 bytes)
```

### Apps

```json
Nope, every app (including nothing - the desktop) has this issue over the internet
```

### Relevant log output

```shell
[2023:02:24:14:00:40]: Info: Compiling shaders...
[2023:02:24:14:00:40]: Info: Compiled shaders
[2023:02:24:14:00:40]: Info: // Testing for available encoders, this may generate errors. You can safely ignore those errors. //
[2023:02:24:14:00:40]: Info: Trying encoder [nvenc]
[2023:02:24:14:00:40]: Info: ddprobe.exe [1] [] returned: 0x00000000
[2023:02:24:14:00:40]: Info: Set GPU preference: 1
[2023:02:24:14:00:40]: Info: 
Device Description : NVIDIA GeForce RTX 3060 Ti
Device Vendor ID   : 0x000010DE
Device Device ID   : 0x00002486
Device Video Mem   : 8038 MiB
Device Sys Mem     : 0 MiB
Share Sys Mem      : 16347 MiB
Feature Level      : 0x0000B100
Capture size       : 1920x1080
Offset             : 0x0
Virtual Desktop    : 3840x1080
[2023:02:24:14:00:40]: Info: Desktop resolution [1920x1080]
[2023:02:24:14:00:40]: Info: Desktop format [DXGI_FORMAT_B8G8R8A8_UNORM]
[2023:02:24:14:00:40]: Info: 
Colorspace         : DXGI_COLOR_SPACE_RGB_FULL_G22_NONE_P709
Bits Per Color     : 8
Red Primary        : [0.639648,0.330078]
Green Primary      : [0.299805,0.599609]
Blue Primary       : [0.150391,0.0595703]
White Point        : [0.3125,0.329102]
Min Luminance      : 0.5 nits
Max Luminance      : 270 nits
Max Full Luminance : 270 nits
[2023:02:24:14:00:40]: Info: SDR color coding [Rec. 601]
[2023:02:24:14:00:40]: Info: Color range: [JPEG]
[2023:02:24:14:00:41]: Info: 
Device Description : NVIDIA GeForce RTX 3060 Ti
Device Vendor ID   : 0x000010DE
Device Device ID   : 0x00002486
Device Video Mem   : 8038 MiB
Device Sys Mem     : 0 MiB
Share Sys Mem      : 16347 MiB
Feature Level      : 0x0000B100
Capture size       : 1920x1080
Offset             : 0x0
Virtual Desktop    : 3840x1080
[2023:02:24:14:00:41]: Info: Desktop resolution [1920x1080]
[2023:02:24:14:00:41]: Info: Desktop format [DXGI_FORMAT_B8G8R8A8_UNORM]
[2023:02:24:14:00:41]: Info: 
Colorspace         : DXGI_COLOR_SPACE_RGB_FULL_G22_NONE_P709
Bits Per Color     : 8
Red Primary        : [0.639648,0.330078]
Green Primary      : [0.299805,0.599609]
Blue Primary       : [0.150391,0.0595703]
White Point        : [0.3125,0.329102]
Min Luminance      : 0.5 nits
Max Luminance      : 270 nits
Max Full Luminance : 270 nits
[2023:02:24:14:00:41]: Info: SDR color coding [Rec. 601]
[2023:02:24:14:00:41]: Info: Color range: [JPEG]
[2023:02:24:14:00:41]: Info: 
Device Description : NVIDIA GeForce RTX 3060 Ti
Device Vendor ID   : 0x000010DE
Device Device ID   : 0x00002486
Device Video Mem   : 8038 MiB
Device Sys Mem     : 0 MiB
Share Sys Mem      : 16347 MiB
Feature Level      : 0x0000B100
Capture size       : 1920x1080
Offset             : 0x0
Virtual Desktop    : 3840x1080
[2023:02:24:14:00:41]: Info: Desktop resolution [1920x1080]
[2023:02:24:14:00:41]: Info: Desktop format [DXGI_FORMAT_B8G8R8A8_UNORM]
[2023:02:24:14:00:41]: Info: 
Colorspace         : DXGI_COLOR_SPACE_RGB_FULL_G22_NONE_P709
Bits Per Color     : 8
Red Primary        : [0.639648,0.330078]
Green Primary      : [0.299805,0.599609]
Blue Primary       : [0.150391,0.0595703]
White Point        : [0.3125,0.329102]
Min Luminance      : 0.5 nits
Max Luminance      : 270 nits
Max Full Luminance : 270 nits
[2023:02:24:14:00:41]: Info: SDR color coding [Rec. 601]
[2023:02:24:14:00:41]: Info: Color range: [JPEG]
[2023:02:24:14:00:41]: Info: 
Device Description : NVIDIA GeForce RTX 3060 Ti
Device Vendor ID   : 0x000010DE
Device Device ID   : 0x00002486
Device Video Mem   : 8038 MiB
Device Sys Mem     : 0 MiB
Share Sys Mem      : 16347 MiB
Feature Level      : 0x0000B100
Capture size       : 1920x1080
Offset             : 0x0
Virtual Desktop    : 3840x1080
[2023:02:24:14:00:41]: Info: Desktop resolution [1920x1080]
[2023:02:24:14:00:41]: Info: Desktop format [DXGI_FORMAT_B8G8R8A8_UNORM]
[2023:02:24:14:00:41]: Info: 
Colorspace         : DXGI_COLOR_SPACE_RGB_FULL_G22_NONE_P709
Bits Per Color     : 8
Red Primary        : [0.639648,0.330078]
Green Primary      : [0.299805,0.599609]
Blue Primary       : [0.150391,0.0595703]
White Point        : [0.3125,0.329102]
Min Luminance      : 0.5 nits
Max Luminance      : 270 nits
Max Full Luminance : 270 nits
[2023:02:24:14:00:41]: Info: SDR color coding [Rec. 601]
[2023:02:24:14:00:41]: Info: Color range: [JPEG]
[2023:02:24:14:00:42]: Info: 
Device Description : NVIDIA GeForce RTX 3060 Ti
Device Vendor ID   : 0x000010DE
Device Device ID   : 0x00002486
Device Video Mem   : 8038 MiB
Device Sys Mem     : 0 MiB
Share Sys Mem      : 16347 MiB
Feature Level      : 0x0000B100
Capture size       : 1920x1080
Offset             : 0x0
Virtual Desktop    : 3840x1080
[2023:02:24:14:00:42]: Info: Desktop resolution [1920x1080]
[2023:02:24:14:00:42]: Info: Desktop format [DXGI_FORMAT_B8G8R8A8_UNORM]
[2023:02:24:14:00:42]: Info: 
Colorspace         : DXGI_COLOR_SPACE_RGB_FULL_G22_NONE_P709
Bits Per Color     : 8
Red Primary        : [0.639648,0.330078]
Green Primary      : [0.299805,0.599609]
Blue Primary       : [0.150391,0.0595703]
White Point        : [0.3125,0.329102]
Min Luminance      : 0.5 nits
Max Luminance      : 270 nits
Max Full Luminance : 270 nits
[2023:02:24:14:00:42]: Info: SDR color coding [Rec. 601]
[2023:02:24:14:00:42]: Info: Color range: [JPEG]
[2023:02:24:14:00:42]: Info: 
Device Description : NVIDIA GeForce RTX 3060 Ti
Device Vendor ID   : 0x000010DE
Device Device ID   : 0x00002486
Device Video Mem   : 8038 MiB
Device Sys Mem     : 0 MiB
Share Sys Mem      : 16347 MiB
Feature Level      : 0x0000B100
Capture size       : 1920x1080
Offset             : 0x0
Virtual Desktop    : 3840x1080
[2023:02:24:14:00:42]: Info: Desktop resolution [1920x1080]
[2023:02:24:14:00:42]: Info: Desktop format [DXGI_FORMAT_B8G8R8A8_UNORM]
[2023:02:24:14:00:42]: Info: 
Colorspace         : DXGI_COLOR_SPACE_RGB_FULL_G22_NONE_P709
Bits Per Color     : 8
Red Primary        : [0.639648,0.330078]
Green Primary      : [0.299805,0.599609]
Blue Primary       : [0.150391,0.0595703]
White Point        : [0.3125,0.329102]
Min Luminance      : 0.5 nits
Max Luminance      : 270 nits
Max Full Luminance : 270 nits
[2023:02:24:14:00:42]: Info: SDR color coding [Rec. 709]
[2023:02:24:14:00:42]: Info: Color range: [JPEG]
[2023:02:24:14:00:42]: Error: Could not open codec [h264_nvenc]: Function not implemented
[2023:02:24:14:00:42]: Info: 
Device Description : NVIDIA GeForce RTX 3060 Ti
Device Vendor ID   : 0x000010DE
Device Device ID   : 0x00002486
Device Video Mem   : 8038 MiB
Device Sys Mem     : 0 MiB
Share Sys Mem      : 16347 MiB
Feature Level      : 0x0000B100
Capture size       : 1920x1080
Offset             : 0x0
Virtual Desktop    : 3840x1080
[2023:02:24:14:00:42]: Info: Desktop resolution [1920x1080]
[2023:02:24:14:00:42]: Info: Desktop format [DXGI_FORMAT_B8G8R8A8_UNORM]
[2023:02:24:14:00:42]: Info: 
Colorspace         : DXGI_COLOR_SPACE_RGB_FULL_G22_NONE_P709
Bits Per Color     : 8
Red Primary        : [0.639648,0.330078]
Green Primary      : [0.299805,0.599609]
Blue Primary       : [0.150391,0.0595703]
White Point        : [0.3125,0.329102]
Min Luminance      : 0.5 nits
Max Luminance      : 270 nits
Max Full Luminance : 270 nits
[2023:02:24:14:00:42]: Info: SDR color coding [Rec. 709]
[2023:02:24:14:00:42]: Info: Color range: [JPEG]
[2023:02:24:14:00:43]: Info: 
Device Description : NVIDIA GeForce RTX 3060 Ti
Device Vendor ID   : 0x000010DE
Device Device ID   : 0x00002486
Device Video Mem   : 8038 MiB
Device Sys Mem     : 0 MiB
Share Sys Mem      : 16347 MiB
Feature Level      : 0x0000B100
Capture size       : 1920x1080
Offset             : 0x0
Virtual Desktop    : 3840x1080
[2023:02:24:14:00:43]: Info: Desktop resolution [1920x1080]
[2023:02:24:14:00:43]: Info: Desktop format [DXGI_FORMAT_B8G8R8A8_UNORM]
[2023:02:24:14:00:43]: Info: 
Colorspace         : DXGI_COLOR_SPACE_RGB_FULL_G22_NONE_P709
Bits Per Color     : 8
Red Primary        : [0.639648,0.330078]
Green Primary      : [0.299805,0.599609]
Blue Primary       : [0.150391,0.0595703]
White Point        : [0.3125,0.329102]
Min Luminance      : 0.5 nits
Max Luminance      : 270 nits
Max Full Luminance : 270 nits
[2023:02:24:14:00:43]: Info: SDR color coding [Rec. 601]
[2023:02:24:14:00:43]: Info: Color range: [JPEG]
[2023:02:24:14:00:43]: Info: 
Device Description : NVIDIA GeForce RTX 3060 Ti
Device Vendor ID   : 0x000010DE
Device Device ID   : 0x00002486
Device Video Mem   : 8038 MiB
Device Sys Mem     : 0 MiB
Share Sys Mem      : 16347 MiB
Feature Level      : 0x0000B100
Capture size       : 1920x1080
Offset             : 0x0
Virtual Desktop    : 3840x1080
[2023:02:24:14:00:43]: Info: Desktop resolution [1920x1080]
[2023:02:24:14:00:43]: Info: Desktop format [DXGI_FORMAT_B8G8R8A8_UNORM]
[2023:02:24:14:00:43]: Info: 
Colorspace         : DXGI_COLOR_SPACE_RGB_FULL_G22_NONE_P709
Bits Per Color     : 8
Red Primary        : [0.639648,0.330078]
Green Primary      : [0.299805,0.599609]
Blue Primary       : [0.150391,0.0595703]
White Point        : [0.3125,0.329102]
Min Luminance      : 0.5 nits
Max Luminance      : 270 nits
Max Full Luminance : 270 nits
[2023:02:24:14:00:43]: Info: SDR color coding [Rec. 601]
[2023:02:24:14:00:43]: Info: Color range: [JPEG]
[2023:02:24:14:00:43]: Warning: nvenc: hevc: replacing nalu prefix data
[2023:02:24:14:00:43]: Info: 
[2023:02:24:14:00:43]: Info: // Ignore any errors mentioned above, they are not relevant. //
[2023:02:24:14:00:43]: Info: 
[2023:02:24:14:00:43]: Info: Found encoder nvenc: [h264_nvenc, hevc_nvenc]
[2023:02:24:14:00:43]: Info: Configuration UI available at [https://localhost:47990]
[2023:02:24:14:00:44]: Info: Registered Sunshine mDNS service
[2023:02:24:14:00:59]: Info: /CN=NVIDIA GameStream Client -- verified
[2023:02:24:14:01:03]: Info: /CN=NVIDIA GameStream Client -- verified
[2023:02:24:14:01:06]: Info: /CN=NVIDIA GameStream Client -- verified
[2023:02:24:14:01:08]: Info: /CN=NVIDIA GameStream Client -- verified
[2023:02:24:14:01:08]: Info: /CN=NVIDIA GameStream Client -- verified
[2023:02:24:14:01:08]: Info: /CN=NVIDIA GameStream Client -- verified
[2023:02:24:14:01:08]: Info: Executing [Desktop]
[2023:02:24:14:01:09]: Info: CLIENT CONNECTED
[2023:02:24:14:01:09]: Info: 
Device Description : NVIDIA GeForce RTX 3060 Ti
Device Vendor ID   : 0x000010DE
Device Device ID   : 0x00002486
Device Video Mem   : 8038 MiB
Device Sys Mem     : 0 MiB
Share Sys Mem      : 16347 MiB
Feature Level      : 0x0000B100
Capture size       : 1920x1080
Offset             : 1920x0
Virtual Desktop    : 3840x1080
[2023:02:24:14:01:09]: Info: Desktop resolution [1920x1080]
[2023:02:24:14:01:09]: Info: Desktop format [DXGI_FORMAT_B8G8R8A8_UNORM]
[2023:02:24:14:01:09]: Info: 
Colorspace         : DXGI_COLOR_SPACE_RGB_FULL_G22_NONE_P709
Bits Per Color     : 8
Red Primary        : [0.639648,0.330078]
Green Primary      : [0.299805,0.599609]
Blue Primary       : [0.150391,0.0595703]
White Point        : [0.3125,0.329102]
Min Luminance      : 0.5 nits
Max Luminance      : 270 nits
Max Full Luminance : 270 nits
[2023:02:24:14:01:09]: Info: Capture format [DXGI_FORMAT_B8G8R8A8_UNORM]
[2023:02:24:14:01:09]: Info: SDR color coding [Rec. 709]
[2023:02:24:14:01:09]: Info: Color range: [MPEG]
[2023:02:24:14:01:18]: Error: Initial Ping Timeout
[2023:02:24:14:01:20]: Info: /CN=NVIDIA GameStream Client -- verified
[2023:02:24:14:01:23]: Info: /CN=NVIDIA GameStream Client -- verified
[2023:02:24:14:01:23]: Info: /CN=NVIDIA GameStream Client -- verified
```

**Comment by KINGLIFER** [comment]
any update on this????

**Comment by silentcovenant** [comment]
I'm unable to connect on lan or over internet. Immediately get the "Connection Terminated" message

**Reply by Cranmoot** [reply]
Experiencing the same as silent covenant...  Had everything working but as of now nothing works.... I'm guessing a microsoft update.    oh, and everything includes:
Steam Deck,  Galaxy S20FE, Google Pixel 5, Galaxy s7,  An older AMD computer running win 10, & a Samsung Tab S

I'll try to gather up the logs as soon as I get a chance...

Quick update SteamLink doesn't appear to function either definitely something windows has broken

**Reply by silentcovenant** [reply]
> Experiencing the same as silent covenant... Had everything working but as of now nothing works.... I'm guessing a microsoft update. oh, and everything includes: Steam Deck, Galaxy S20FE, Google Pixel 5, Galaxy s7, An older AMD computer running win 10, & a Samsung Tab S
> 
> I'll try to gather up the logs as soon as I get a chance...
> 
> Quick update SteamLink doesn't appear to function either definitely something windows has broken

Actually, I figured out my issue. It was one of my audio devices that needed to be reset in Sunshine. My Virtual Sink device needed to be changed under the Configuration > Audio/Video > Virtual Sink

**Reply by Cranmoot** [reply]
Good Call on this one silentcovenant.... checked my audio sink devices aand they had indeed chagned their adresses, Strange.  Thanks for the insight.

**Reply by adelagon** [reply]
Nice catch on this. I can confirm that updating the virtual sink fixes the issue. Just run the tools\audio-info.exe and update sunshine config accordingly.

**Reply by borland502** [reply]
Sorry to necro this thread, but if anyone comes across this and the above does not work then it is the ping timeout that may be involved if you check the logs.  A permanent solution is a WIP but the issue may be that I suppress ipv6 with prejudice on my network and ping is attempting to respond over ipv6.  For now, I just set the timeout to a lot o' digits.

**Comment by zmk5566** [comment]
Hey I was having the same issue today. 

@PicoPlanetDev 

I think the error occurred in  Error: Initial Ping Timeout

I tried to read the source code in the file stream.cpp, it seems that it is trying to use a ping mechanism as a heartbeat connection check.

https://github.com/LizardByte/Sunshine/blob/09dff341056a26c1a41121a068327a3f8d9616df/src/stream.cpp#L1312

in Line 1312,  the receive ping function try to use PING to sense the existence of the client.

Since ping is using ICMP which will create hassle during port forwarding, maybe instead of using ping, a heartbeat message through UDP with a response mechanism can perform the same functionality?

I had a very brief read of the code, hope to hear from you soon.

**Reply by sigmondkukla** [reply]
That might be the culprit! As I'm still having this issue, I'd be very interested if anyone knows how to implement it.

**Reply by lwd369** [reply]
As a temporary and crude solution, I have compiled a version that ignores ping responses. [It](https://github.com/lwd369/Sunshine/commit/215d912395d52d975618f5756cfd3293e9cb6a33) simply disregards the error value returned by the `recv_ping` function. While this solution may cause issues in an unstable network environment, it can temporarily address situations wherein streaming via the internet is impossible.
[sunshine-windows-installer.zip](https://github.com/LizardByte/Sunshine/files/11303204/sunshine-windows-installer.zip)

**Reply by ReenigneArcher** [reply]
Just increase this value. https://docs.lizardbyte.dev/projects/sunshine/en/latest/about/advanced_usage.html#ping-timeout

**Reply by lwd369** [reply]
> Just increase this value. https://docs.lizardbyte.dev/projects/sunshine/en/latest/about/advanced_usage.html#ping-timeout

I tried increasing the `ping_timeout` to 2 hours, but while the game I initially streamed worked fine, later attempts to resume streaming resulted in being unable to start the stream and getting stuck. To continue playing, I had to start streaming the desktop application.

**Comment by IM-001** [comment]
I'm having the same issue, I tried to update my Virtual Sink in the devices with the Steam Streaming Speakers Device ID and it didn't work.

My setup is the following: I am trying to play RetroBat (RetroArch frontend) on my Nvidia Shield with Sunshine / Moonlight.  I have 3 apps / links on my sunshine computer: Desktop, Retrobat and Steam.  So if I try to play the Retrobat app, it crashes after 5 secs with "Connection Terminated". 

My funny workaround is I launch the Steam app on moonlight, choose the option Exit Steam (so it closes steam on my desktop, but remain connected), and then it switches automatically to the next opened app on my desktop which is RetroBat.  I can then play my games like that.  But whatever I do I get disconnected if I choose the RetroBat app on moonlight.

**Comment by DaddyDong** [comment]
I'm having the exact same issue as everyone else. Tried updating the "Virtual Sink" and still the same issue. It was working perfect until my windows updated. I get disconnected after a few seconds of being connected and get a "Connection Terminated" error. When I check the Sunshine cmd prompt as to why it disconnected I get an "Error: Initial Ping Timeout". This is getting extremely frustrating....

**Comment by zachsirotto** [comment]
I'm also seeing this issue after updating windows last night.

**Comment by cgutman** [comment]
The virtual sink and audio sink issues have been fixed in the latest nightly build. You should specify the devices by name rather than ID in nightly and v0.20.0+ (when it's released).

Nightly downloads are here if you want to try them out: https://github.com/LizardByte/Sunshine/releases/tag/nightly-dev

**Reply by prog-amateur2** [reply]
Hello thank you very much for your advise. I discovered sunshine a few time ago (it is not so popular), it seems there are several documentation websites, and most of the tutorials are for Windows. 

I am having the same issue as the author of this discussion (default setup, moonlight connecting well, then a black screen with `connection terminated` error) but my setting is : 

* **OS** : debian sid
* **Sunshine** : flatpak version 0.19.1 (7 weeks ago)
* **Tools for audio-info** : could not find them
 
So my questions are : 
1. From your link (nightly builds), how can I overwrite Sunshine v0.19.1 with the nightly flatpak file (is there a possibility of corruption) ?
2. How can I find the device names by ID on Debian ? 

Thank you very much for your kind help !

**Reply by ReenigneArcher** [reply]
There's only one documentation website 👉 https://docs.lizardbyte.dev/projects/sunshine

> How can I find the device names by ID on Debian ?

https://docs.lizardbyte.dev/projects/sunshine/en/latest/about/advanced_usage.html#audio

**Comment by phamaralbr** [comment]
I'm having the same problem. Sunshine works fine locally, but throws the timeout error when I access it from an external network.

```
[2023:07:12:09:15:46]: Warning: Rejected connection from [XXX.XXX.XXX.XXX]: it's not properly set up
[2023:07:12:09:15:56]: Error: Initial Ping Timeout
[2023:07:12:09:15:56]: Info: XXX.XXX.XXX.XXX: Ping Timeout
```

I have tried the latest stable (0.20.0) and the nightly. Problem persists

**Comment by Matgoeth** [comment]
Unfortunately have the same issue:

> [2023:07:21:09:45:15]: Info: // Ignore any errors mentioned above, they are not relevant. // [2023:07:21:09:45:15]: Info: [2023:07:21:09:45:15]: Info: Found encoder nvenc: [h264_nvenc, hevc_nvenc] [2023:07:21:09:45:26]: Error: Initial Ping Timeout [2023:07:21:09:45:26]: Error: Initial Ping Timeout [2023:07:21:09:45:26]: Info: xxx.xxx.xxx.xxx: Ping Timeout

Happy to provide logs/configs if needed, using latest stable (0.20.0) and Moonlight (4.3.1), streaming via NVIDIA Experience (mstsc) seem
to be fine on same moonlight

**Reply by andrew-kennedy** [reply]
I'm having the same issue on 0.20.0 and moonlight 4.3.1. Internal network works great, external network works for 5 seconds until the "initial ping timeout" hits.

**Comment by Kolcha2009** [comment]
Alright guys, I have found the solution of this problem, at least in my case it was the router settings.
I am very new to sunshine, since I have used regular gamestream from nvidia and directly got the ignitial ping error. Now I tried to delete all my port settings and restartet the router because sunshine is using upnp and suddenly everything worked perfect.

I hope this will help somebody too because it drives me crazy :)

**Reply by ReenigneArcher** [reply]
Probably better to just manually port forward what you need.

**Reply by Kolcha2009** [reply]
Maybe, but since I have cleared everything from port forwording, that I have needed for nvidia gamestream and let upnp do its thong, everything works like a charm. But you have to know that ;)

**Comment by tarruda** [comment]
I still have the "initial ping timeout" issue. I don't enable uPnP because it has security issues: https://www.upguard.com/blog/what-is-upnp . Instead, I manually forward all the required ports to a separate VPN which allows me to securely connect through the internet.

What happens is that streaming works fine in the LAN, but when access through the port-forwarded VPN I get the "Connection Terminated" message and I see the "Error: Initial Ping Timeout" in Sunshine logs.

Can we add a configuration option to disable disconnection if the ping fails?

**Reply by ReenigneArcher** [reply]
I think there's an adjustment in nightly for this. Can you try it?

**Reply by tarruda** [reply]
Ignore this. It seems I was forwarding the wrong ports. After forwarding:

47984,47989,48010 (TCP)
47998-48000 (UDP)

Everything seems to be working perfectly.

**Reply by ReenigneArcher** [reply]
The ports required to be forwarded are listed in the config UI.

**Reply by Gigantorz** [reply]
Hey Tarruda, can you explain what happened and how you fixed it? I think I might be dealing with the same issue.

**Reply by tarruda** [reply]
@Gigantorz I was simply forwarding the wrong ports. If you forward:

47984,47989,48010 (TCP)
47998-48000 (UDP)

Everything should work

**Comment by pietergmail** [comment]
Also running into this issue when connecting remotely through a WireGuard vpn to a different subnet, and then port-forwarding: 47800 47998 47999 48010 4989 47984 on a windows pc. Running sunshine v0.23.1 on windows. I have no issues with connecting through my steamdeck while on the same subnet.

[a comment by an automated account is not shown]
